File: python/src/runapi/openai_tts/contract_gen.py
CONTRACT = {
    "text-to-speech": {
        "models": ["tts-1", "tts-1-hd"],
        "fields_by_model": {
            "tts-1": {
                "model": {
                    "required": True
                },
                "text": {
                    "required": True,
                    "max": 4096,
                    "length": True
                }
            },
            "tts-1-hd": {
                "model": {
                    "required": True
                },
                "text": {
                    "required": True,
                    "max": 4096,
                    "length": True
                }
            }
        }
    }
}

File: ruby/lib/runapi/openai_tts/contract_gen.rb
# frozen_string_literal: true

module RunApi
  module OpenaiTts
    CONTRACT = {
      "text-to-speech" => {
        "models" => ["tts-1", "tts-1-hd"],
        "fields_by_model" => {
          "tts-1" => {
            "model" => {
              "required" => true
            },
            "text" => {
              "required" => true,
              "max" => 4096,
              "length" => true
            }
          },
          "tts-1-hd" => {
            "model" => {
              "required" => true
            },
            "text" => {
              "required" => true,
              "max" => 4096,
              "length" => true
            }
          }
        }
      }
    }.freeze
  end
end
